pkg/serviceaccount/jwt_test.go

```diff
@@ -342,7 +343,11 @@ func TestTokenGenerateAndValidate(t *testing.T) {
 				return tc.Client.CoreV1().Pods(namespace).Get(context.TODO(), name, metav1.GetOptions{})
 			})),
 		)
-		authn := serviceaccount.JWTTokenAuthenticator([]string{serviceaccount.LegacyIssuer, "bar"}, tc.Keys, auds, serviceaccount.NewLegacyValidator(tc.Client != nil, getter, nil))
+		var secretsWriter typedv1core.SecretsGetter
+		if tc.Client != nil {
+			secretsWriter = tc.Client.CoreV1()
+		}
+		authn := serviceaccount.JWTTokenAuthenticator([]string{serviceaccount.LegacyIssuer, "bar"}, tc.Keys, auds, serviceaccount.NewLegacyValidator(tc.Client != nil, getter, secretsWriter))
```

Comment:
if `nil` gets passed in here for `secretsWriter`, what happens if the feature is enabled?

Reply:
the test will crash if lookup is true. maybe worthy adding sanity checks for getter and secretsWriter.

Reply:
yeah, maybe check those are non-nil if they would be used at runtime, and plumb an error back up from NewLegacyValidator so we catch problems at init time

Reply:
done